backend/typescript/prisma/schema.prisma

```diff
@@ -0,0 +1,43 @@
+model Task {
+    id        Int      @id @default(autoincrement())
+    category  Categories 
+    title     String   
+    status    Statuses
+    description String
+
+    // TODO: CHANGE THESE AFTER USER MODELS HAVE BEEN MADE
+    assignee String
+    assigner String
+
+    credit_value Float
+    start_date DateTime
+    end_date DateTime?
+    comments String
+    recurrence_frequency String?
+    // recurrence_days String // store days a comma separted string to be parsed for usage
+    warnings Warning[]
+}
+
+enum Categories {
+    HOUSEHOLD
+    CHORES
+    FINANCE
+    // other categories
+}
+
+enum Statuses {
+    PENDING_APPROVAL
+    INCOMPLETE
+    COMPLETE
+    EXCUSED
+}
+
+model Warning {
+    resident_id Int @id @default(autoincrement())
```

Comment:
Warnings should have their own `id` that is set as `@id`.
The field that is set as `@id` must be unique so having it this way means that each resident can only have up to 1 warning which isn't true.
Instead set up a different `id` field as `@id`, then `resident_id` will need to be setup as a relation similar to `related_task` at the bottom of this file.